src/python_optimiser_interface.cpp

```diff
@@ -871,6 +871,70 @@ extern "C"
     return PyBool_FromLong(optimiser->refine_partition);
   }
 
+  PyObject* _Optimiser_set_max_comm_size(PyObject *self, PyObject *args, PyObject *keywds)
+  {
+    PyObject* py_optimiser = NULL;
+    unsigned int max_comm_size = 0;
+    static char* kwlist[] = {"optimiser", "max_comm_size", NULL};
+
+    #ifdef DEBUG
+      cerr << "Parsing arguments..." << endl;
+    #endif
+
+    if (!PyArg_ParseTupleAndKeywords(args, keywds, "OI", kwlist,
+                                     &py_optimiser, &max_comm_size))
+        return NULL;
+
+    #ifdef DEBUG
+      cerr << "set_max_comm_size(" << max_comm_size << ");" << endl;
+    #endif
+
+    #ifdef DEBUG
+      cerr << "Capsule optimiser at address " << py_optimiser << endl;
+    #endif
+    Optimiser* optimiser = decapsule_Optimiser(py_optimiser);
+    #ifdef DEBUG
+      cerr << "Using optimiser at address " << optimiser << endl;
+    #endif
+
+    optimiser->max_comm_size = max_comm_size;
+
+    Py_INCREF(Py_None);
+    return Py_None;
+  }
+
+  PyObject* _Optimiser_get_max_comm_size(PyObject *self, PyObject *args, PyObject *keywds)
+  {
+    PyObject* py_optimiser = NULL;
+    static char* kwlist[] = {"optimiser", NULL};
+
+    #ifdef DEBUG
+      cerr << "Parsing arguments..." << endl;
+    #endif
+
+    if (!PyArg_ParseTupleAndKeywords(args, keywds, "O", kwlist,
+                                     &py_optimiser))
+        return NULL;
+
+    #ifdef DEBUG
+      cerr << "get_max_comm_size();" << endl;
+    #endif
+
+    #ifdef DEBUG
+      cerr << "Capsule optimiser at address " << py_optimiser << endl;
+    #endif
+    Optimiser* optimiser = decapsule_Optimiser(py_optimiser);
+    #ifdef DEBUG
+      cerr << "Using optimiser at address " << optimiser << endl;
+    #endif
+
+    #ifdef IS_PY3K
+    return PyLong_FromLong(optimiser->max_comm_size);
```

Comment:
Here, to translate from a `size_t` you need `PyLong_FromSize_t`, this works for both Python 2 and Python 3.

Reply:
Just to make sure - this means I should use `PyLong_FromSize_t` w/o an `#ifdef`?

Reply:
Good point that elsewhere it could also use `PyLong_FromLong`, not sure why that is still in there. Simply use unconditionally `PyLong_FromSize_t` should be fine, thanks!